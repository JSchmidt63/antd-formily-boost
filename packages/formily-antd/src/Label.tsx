import { useField } from '@formily/react';
import { observer } from '@formily/reactive-react';
import React, { Fragment } from 'react';
type LabelProps = {
    value: string;
    prefix?: string;
    suffix?: string;
};

const Label: React.FC<LabelProps> = observer((props) => {
    const field = useField();
    let result = '';
    if (props.prefix) {
        result += props.prefix;
    }
    result += props.value;
    if (props.suffix) {
        result += props.suffix;
    }
    return <Fragment>{result}</Fragment>;
});

export default Label;
