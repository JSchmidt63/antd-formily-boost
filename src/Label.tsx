import { useField } from '@formily/react';
import { observer } from '@formily/reactive-react';
import React, { Fragment } from 'react';
type LabelProps = {
    value: string;
    prefix?: string;
    suffix?: string;
};

const Label: React.FC<LabelProps> = observer((props) => {
    let result = '';
    if (props.prefix) {
        result += props.prefix;
    }
    let value = props.value;
    if (value == undefined) {
        value = '';
    }
    result += value;
    if (props.suffix) {
        result += props.suffix;
    }
    return <Fragment>{result}</Fragment>;
});

export default Label;
