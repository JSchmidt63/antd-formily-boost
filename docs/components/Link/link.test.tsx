import Label from './label';
import { render } from '@testing-library/react';
import React from 'react';

beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
        value: jest.fn(() => {
            return {
                matches: true,
                addListener: jest.fn(),
                removeListener: jest.fn(),
            };
        }),
    });
});

test('Label', () => {
    const { asFragment } = render(<Label />);
    expect(asFragment()).toMatchSnapshot();
});
