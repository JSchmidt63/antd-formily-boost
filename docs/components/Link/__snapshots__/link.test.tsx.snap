// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label 1`] = `
<DocumentFragment>
  <div
    class="ant-formily-layout"
  >
    <form>
      <div>
        {"data":[]}
      </div>
    </form>
  </div>
</DocumentFragment>
`;
